### PR TITLE
[risk=no][RW=5014]Initial react router implementation

### PR DESCRIPTION
**Pull request by petesantos**
Description:

Create initial implementation of routing component and convert one route (cookie-policy) to use the new router

Change can be tested here:
https://psantos-dot-all-of-us-workbench-test.appspot.com/cookie-policy

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI + API server with `yarn test-local`
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -346,6 +341,10 @@ const routes: Routes = [
         data: {title: 'Create Workspace', mode: WorkspaceEditMode.Create}
       }
     ]
+  }, {
+    path: '**',
+    component: AppRouting,
+    data: {title: 'Sign In'}
```
Comment (jmthibault79) [inline ui/src/app/app-routing.module.ts:347]: why "Sign In" ?

File: ui/src/app/components/app-router.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  const routeParams = useParams();
+  const routeHistory = useHistory();
+
+  return <Route path={path} >
+    <Component urlParams={routeParams} routeHistory={routeHistory} routeConfig={data}/>
+  </Route>;
+};
+
+export const ProtectedRoutes = (
+  {path, guards, children}: {path: string, guards: Guards, children: any[] }): React.ReactElement => {
```
Comment (jmthibault79) [inline ui/src/app/components/app-router.tsx:34]: can children be more strongly typed here?  Fragment[] perhaps?

File: ui/src/app/app-routing.module.ts (hunk 1 above)
Comment (petesantos) [reply ui/src/app/app-routing.module.ts]: I left this in accidentally - removed - thanks!

File: ui/src/app/components/app-router.tsx (hunk 2 above)
Comment (petesantos) [reply ui/src/app/components/app-router.tsx]: I did mean to update this - I set it to React.ReactNode. That should cover any type of child that react accepts

**Pull request merged**